lib/iris/fileformats/pp.py

```diff
@@ -1297,6 +1301,16 @@ def core_data(self):
         return self._data
 
     @property
+    def data_dtype(self):
+        return self._data.dtype \
+            if self._data_dtype is None \
+            else self._data_dtype
```

Comment:
Hang on, won't this always return `self._data_dtype`, whether it's `None` or not??